src/Backups/BackupEntriesCollector.cpp

```diff
@@ -781,6 +781,17 @@ void BackupEntriesCollector::makeBackupEntriesForTablesDefs()
 
         const String & metadata_path_in_backup = table_info.metadata_path_in_backup;
         backup_entries.emplace_back(metadata_path_in_backup, std::make_shared<BackupEntryFromMemory>(new_create_query->formatWithSecretsOneLine()));
+
+        // metadata_version only makes sense for replicated tables.
+        // Process metadata_version separately from the parts
+        // to ensure it's captured regardless of part processing (e.g. structure_only=1)
+        if (table_info.storage->getName().starts_with("Replicated"))
+        {
+            int32_t metadata_version = table_info.storage->getInMemoryMetadataPtr()->metadata_version;
+            backup_entries.emplace_back(
```

Comment:
Not sure why we don't care about locks here.
This is original behavior anyway, so we don't make anything worse.